File: 3453-generate-binary-strings-without-adjacent-zeros/generate-binary-strings-without-adjacent-zeros.cpp
class Solution {
public:

    void solve(int i, string curr, vector<string>&ans, int n, bool canInclude0){
        if(i == n){
            ans.push_back(curr);
            return;
        }
        if(canInclude0){
            curr+='0';
            solve(i+1,curr,ans,n,false);
            curr.pop_back();
            curr+='1';
            solve(i+1,curr,ans,n,true);
            curr.pop_back();
        }else{
            curr+='1';
            solve(i+1,curr,ans,n,true);
            curr.pop_back();
        }
    }

    vector<string> validStrings(int n) {
        vector<string>ans;
        string current;
        solve(0,current,ans,n,true);
        return ans;
    }
};

File: 3453-generate-binary-strings-without-adjacent-zeros/generate-binary-strings-without-adjacent-zeros.py
class Solution:
    def validStrings(self, n: int) -> List[str]:
        def solve(i, curr, ans, n, canInclude0):
            if i == n:
                ans.append(curr)
                return
            if canInclude0 is True:
                curr+='0'
                solve(i+1,curr,ans,n,False)
                curr = curr[:-1]
                curr+='1'
                solve(i+1,curr,ans,n,True)
                curr = curr[:-1]
            else:
                curr+='1'
                solve(i+1,curr,ans,n,True)
                curr = curr[:-1]
        ans = []
        curr = ""
        solve(0,curr,ans,n,True)
        return ans